flatpak/org.eclipse.theia.yaml

```diff
@@ -0,0 +1,91 @@
+app-id: org.eclipse.theia
+base: org.electronjs.Electron2.BaseApp
+base-version: '25.08'
+runtime: org.freedesktop.Sdk
+runtime-version: '25.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node24
+command: theia
+finish-args:
+  - --allow=devel
+  - --device=all
+  - --env=ELECTRON_TRASH=gio
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=host
+  - --persist=.theia-ide
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=ssh-auth
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --talk-name=org.freedesktop.Flatpak # Required to run commands in the host system.
+  - --talk-name=org.freedesktop.secrets # Access to authentication services, e.g. for git.
+separate-locales: false
+modules:
+  - name: theia
+    sources:
+      - generated-sources.json
+      - type: dir
+        path: ..
+      - type: patch
+        path: electron-builder.patch
+      # Bundled VS Code extensions - update with: ./update-extensions.sh
+      - extension-sources.json
+      - type: script
+        dest-filename: wrapper.sh
+        commands:
+          # Escape the Flatpak sandbox to run on the host for access to SDKs and tools
```

Comment:
It should probably be emphasized that this is a hack. I'm not sure if there's an upstream tracking issue for running with this kind of host access.

Reply:
No, Flatpak devs are adamant: the sandbox is non-negotiable. I agree it's hackish but the Zed editor happens to be distributed on Flathub, so it's fine.

Reply:
Well, it's an obvious hack. And as such, it'll only be accepted onto Flathub with official backing: https://docs.flathub.org/docs/for-app-authors/requirements#submissions-incompatible-or-restricted-by-flatpak-sandbox